tests/serdetest.c

```diff
@@ -48,4 +49,213 @@ TEST (musig2, commitments_serialize_deserialize) {
+    ASSERT_EQ(err, MUSIG2_OK);
+
+    // Fuzz a public key of a randomly selected signer.
+    int fuzz_index = rand() % NR_SIGNERS;
+    musig2_helper_fuzz_data(&mcs_list[fuzz_index].keypair.data[MUSIG2_SCALAR_BYTES], MUSIG2_PUBKEY_BYTES_FULL);
+
+    err = musig2_helper_sign(mcs_list, mps, NR_SIGNERS);
+    ASSERT_EQ(err, MUSIG2_OK);
+
+    err = musig2_aggregate_partial_sig(mps, signature, NR_SIGNERS);
+    ASSERT_EQ(err, MUSIG2_OK);
+
+    // Verification should fail since one of the signers' signature is generated with an incorrect public key.
+    err = musig2_helper_verify(serialized_pubkey_list, signature, MSG_1, MSG_1_LEN, NR_SIGNERS);
+    ASSERT_EQ(err, MUSIG2_INVALID);
+}
+
+/* Set up the verifier given a list of public keys including a fuzzed public key in it.
+ * Try to verify the aggregate signature. */
+TEST (musig2, fuzz_pubkey_verify_aggregate) {
+
+    secp256k1_context *ctx = secp256k1_context_create(SECP256K1_CONTEXT_VERIFY);
+    musig2_context_signer mcs_list[NR_SIGNERS]; // Array that holds NR_SIGNERS musig2_context_signer
+    musig2_context_signature mps[NR_SIGNERS];
+    unsigned char serialized_batch_list[NR_MESSAGES * NR_SIGNERS * V * MUSIG2_PUBKEY_BYTES_COMPRESSED];
+    unsigned char serialized_pubkey_list[NR_SIGNERS * MUSIG2_PUBKEY_BYTES_COMPRESSED];    // Signers' public key list
+    unsigned char signature[MUSIG2_BYTES];
+    MUSIG2_ERROR err;
+
+    err = musig2_helper_setup(mcs_list, serialized_pubkey_list, serialized_batch_list, NR_SIGNERS);
+    ASSERT_EQ(err, MUSIG2_OK);
+
+    err = musig2_helper_precomputation(serialized_pubkey_list, serialized_batch_list, mcs_list, NR_SIGNERS);
+    ASSERT_EQ(err, MUSIG2_OK);
+
+    err = musig2_helper_sign(mcs_list, mps, NR_SIGNERS);
+    ASSERT_EQ(err, MUSIG2_OK);
+
+    err = musig2_aggregate_partial_sig(mps, signature, NR_SIGNERS);
+    ASSERT_EQ(err, MUSIG2_OK);
+
+    int fuzz_index = (rand() % NR_SIGNERS) * MUSIG2_PUBKEY_BYTES_COMPRESSED;
+    musig2_helper_fuzz_data(&serialized_pubkey_list[fuzz_index], MUSIG2_PUBKEY_BYTES_COMPRESSED);
+
+    err = musig2_helper_verify(serialized_pubkey_list, signature, MSG_1, MSG_1_LEN, NR_SIGNERS);
+
+    switch(err){
+        case MUSIG2_ERR_PARSE_PK:
+            secp256k1_pubkey pubkey;
+            int res;
+            res = secp256k1_ec_pubkey_parse(ctx, &pubkey, &serialized_pubkey_list[fuzz_index], MUSIG2_PUBKEY_BYTES_COMPRESSED);
+            ASSERT_EQ(res, 0);
+            break;
+        case MUSIG2_INVALID:
+            int cmp_res;
+            musig2_aggr_pubkey aggr_pubkey, fuzz_aggr_pubkey;
+            assert(musig2_prepare_verifier(&aggr_pubkey, serialized_pubkey_list, NR_SIGNERS));
+            assert(secp256k1_xonly_pubkey_from_pubkey(ctx, &fuzz_aggr_pubkey, NULL, &mcs_list[0].mc.aggr_pubkey));
+            cmp_res = secp256k1_xonly_pubkey_cmp(ctx, &fuzz_aggr_pubkey, &aggr_pubkey);
+            ASSERT_NE(cmp_res, 0);
+            break;
+        default:
+            ASSERT_NE(err, MUSIG2_OK);
```

Comment:
These two cases can be considered in a single case, and you would check the expected behaviour, mainly that: 
1. deserialisation of the fuzzed public key is correct
2. `musig2_prepare_verifier` is correct
3. `musig2_verify` fails